cmake/alpakaCommon.cmake

```diff
@@ -261,278 +261,146 @@ endif()
-                    set(CUDA_SEPARABLE_COMPILATION ON)
-                endif()
+            # NOTE: Since CUDA 10.2 this option is also alternatively called '--extended-lambda'
+            if(ALPAKA_CUDA_NVCC_EXPT_EXTENDED_LAMBDA)
+                target_compile_options(alpaka INTERFACE $<$<COMPILE_LANGUAGE:CUDA>:--expt-extended-lambda>)
+            endif()
 
-                # nvcc sets no linux/__linux macros on OpenPOWER linux
-                # nvidia bug id: 2448610
-                if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
-                    if(CMAKE_SYSTEM_PROCESSOR STREQUAL "ppc64le")
-                        list(APPEND CUDA_NVCC_FLAGS -Dlinux)
-                    endif()
-                endif()
+            # This is mandatory because with c++14 many standard library functions we rely on are constexpr (std::min, std::multiplies, ...)
+            target_compile_options(alpaka INTERFACE $<$<COMPILE_LANGUAGE:CUDA>:--expt-relaxed-constexpr>)
 
-                # NOTE: Since CUDA 10.2 this option is also alternatively called '--extended-lambda'
-                if(ALPAKA_CUDA_NVCC_EXPT_EXTENDED_LAMBDA)
-                    list(APPEND CUDA_NVCC_FLAGS --expt-extended-lambda)
+            if((CMAKE_BUILD_TYPE STREQUAL "Debug") OR (CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo"))
+                if(MSVC)
+                    message(WARNING "${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION} does not support -G with CUDA! "
+                                    "Device debug symbols NOT added.")
+                else()
+                    target_compile_options(alpaka INTERFACE $<$<COMPILE_LANGUAGE:CUDA>:-G>)
                 endif()
-                # This is mandatory because with c++14 many standard library functions we rely on are constexpr (std::min, std::multiplies, ...)
-                list(APPEND CUDA_NVCC_FLAGS --expt-relaxed-constexpr)
-
-                foreach(_CUDA_ARCH_ELEM ${ALPAKA_CUDA_ARCH})
-                    # set flags to create device code for the given architecture
-                    list(APPEND CUDA_NVCC_FLAGS
-                        --generate-code=arch=compute_${_CUDA_ARCH_ELEM},code=sm_${_CUDA_ARCH_ELEM}
-                        --generate-code=arch=compute_${_CUDA_ARCH_ELEM},code=compute_${_CUDA_ARCH_ELEM}
-                    )
-                endforeach()
+            endif()
 
-                if(NOT MSVC)
-                    list(APPEND CUDA_NVCC_FLAGS -std=c++${ALPAKA_CXX_STANDARD})
-                endif()
+            if(ALPAKA_CUDA_FAST_MATH)
+                target_compile_options(alpaka INTERFACE $<$<COMPILE_LANGUAGE:CUDA>:--use_fast_math>)
+            endif()
 
-                set(CUDA_HOST_COMPILER ${CMAKE_CXX_COMPILER})
+            if(ALPAKA_CUDA_FTZ)
+                target_compile_options(alpaka INTERFACE $<$<COMPILE_LANGUAGE:CUDA>:--ftz=true>)
+            else()
+                target_compile_options(alpaka INTERFACE $<$<COMPILE_LANGUAGE:CUDA>:--ftz=false>)
+            endif()
```

Comment:
Without being a CMake expert: Can we refactor this into something like:
```
 target_compile_options(alpaka INTERFACE $<$<COMPILE_LANGUAGE:CUDA>:--ftz=${ALPAKA_CUDA_FTZ}>)
```

Reply:
`ALPAKA_CUDA_FTZ` is currently either `ON` or `OFF` and not `true`/`false`. There is probably a nice way to write this if-else as generator expression but this is a purely stylistic change that is not necessary for this PR. If someone has time to refactor this, please provide the proposed syntax.